plugins/hls-refactor-plugin/test/Main.hs

```diff
@@ -2388,7 +2388,7 @@ addTypeAnnotationsToLiteralsTest = testGroup "add type annotations to literals t
                , ""
                , "f = 1"
                ])
-#if MIN_VERSION_ghc(9,4,0)
```

Comment:
Do you see any issue with this approach?
Instead of using MIN_VERSION_{package} macros defined by cabal, I switched to using [MIN_VERSION_GLASSWOG_HASKELL](https://downloads.haskell.org/ghc/latest/docs/users_guide/phases.html?highlight=min_version_glasgow_haskell#standard-cpp-macros) macro defined by GHC.
This allows us to remove unused (reported by -Wunused-packages) ghc dependency from test suite.

Reply:
I _think_ this should be fine, wdyt @wz1000 ?

Reply:
To be clear, multiple approaches are used across the codebase. 
Clearly the advantage of the first 2 mentioned below is that they don't require unused dependency on GHC.
But there might be issues with them that I'm not aware of.

Currently on master:

The form #if __GLASGOW_HASKELL__ >= XYZ is used 11x
The form `#if MIN_VERSION_GLASGOW_HASKELL(W,X,Y,Z)` is used 4x
The form `#if MIN_VERSION_ghc(X,Y,Z)` is used 354x

Reply:
So I guess one way of looking at it is that that the first two condition on the version of GHC-the-compiler, whereas the latter conditions of the version of ghc-the-library. Arguably it's slightly more semantically correct to give conditionals based on ghc-the-library in most cases, since it's differences between versions of that library that cause problems. 

In this case it is _also_ true that the difference is arguably caused by a difference in (our transitive dependency) ghc-the-library, but we can't _say_ that without depending on it directly. In that sense using one of the first two forms is a bit of a hack that relies on the fact that the version of ghc-the-library matches the version of GHC-the-compiler. I have no idea if those can ever come apart, but I guess it's semantically a bit weird?

Reply:
Also! I'm not sure we even need the CPP! We have a value `ghcVersion` that is defined (with CPP) to give us the GHC version. So we can just use that here: `if ghcVersion >= GHC94 then ... else ...`.

Reply:
Ok, in this particular case I switched to ghcVersion as I think no-CPP solution is generally preferable if it can be made to work.